Compile fails on lock_utils.c: 127:14: error: variable has incomplete type 'union semun'

Hi,

I get the following error when trying to compile on FreeBSD 13 against PostgreSQL 14.

```
[root@pgsql1 ~]# pkg info|grep postgres
postgresql14-client-14.2       PostgreSQL database (client)
postgresql14-contrib-14.2      The contrib utilities from the PostgreSQL distribution
postgresql14-server-14.2       PostgreSQL is the most advanced open-source database available anywhere

[root@pgsql1 /tmp/pg_auto_failover]# uname -a
FreeBSD pgsql1 13.1-PRERELEASE FreeBSD 13.1-PRERELEASE #0 stable/13-n249838-e8bb416da13: Thu Mar  3 02:31:18 UTC 2022     root@releng3.nyi.freebsd.org:/usr/obj/usr/src/amd64.amd64/sys/GENERIC  amd64

[root@pgsql1 /tmp/pg_auto_failover]# cc -v
FreeBSD clang version 13.0.0 (git@github.com:llvm/llvm-project.git llvmorg-13.0.0-0-gd7b669b3a303)
Target: x86_64-unknown-freebsd13.1
Thread model: posix
InstalledDir: /usr/bin


gmake[2]: Entering directory '/tmp/pg_auto_failover/src/bin/pg_autoctl'
cc -std=c99 -D_GNU_SOURCE -g -I /usr/local/include -I /usr/local/include/postgresql/server -I /usr/local/include/postgresql/internal -Wall -Wmissing-prototypes -Wpointer-arith -Wdeclaration-after-statement -Werror=vla -Werror=unguarded-availability-new -Wendif-labels -Wmissing-format-attribute -Wformat-security -fno-strict-aliasing -fwrapv -Wno-unused-command-line-argument -O2 -pipe  -fstack-protector-strong -fno-strict-aliasing  -Wformat -Wall -Werror=implicit-int -Werror=implicit-function-declaration -Werror=return-type -Wno-declaration-after-statement -Wno-missing-braces -I/tmp/pg_auto_failover/src/bin/pg_autoctl/../lib/pg -I/tmp/pg_auto_failover/src/bin/pg_autoctl/../lib/log/src/ -I/tmp/pg_auto_failover/src/bin/pg_autoctl/../lib/subcommands.c/ -I/tmp/pg_auto_failover/src/bin/pg_autoctl/../lib/libs/ -I/tmp/pg_auto_failover/src/bin/pg_autoctl/../lib/parson/  -c -MMD -MP -MF/tmp/pg_auto_failover/src/bin/pg_autoctl//.deps/lock_utils.Po -o lock_utils.o lock_utils.c
lock_utils.c:127:14: error: variable has incomplete type 'union semun'
        union semun semun;
                    ^
lock_utils.c:127:8: note: forward declaration of 'union semun'
        union semun semun;
              ^
lock_utils.c:199:14: error: variable has incomplete type 'union semun'
        union semun semun;
                    ^
lock_utils.c:199:8: note: forward declaration of 'union semun'
        union semun semun;
              ^
2 errors generated.
gmake[2]: *** [Makefile:69: lock_utils.o] Error 1
gmake[2]: Leaving directory '/tmp/pg_auto_failover/src/bin/pg_autoctl'
gmake[1]: *** [Makefile:7: pg_autoctl] Error 2
gmake[1]: Leaving directory '/tmp/pg_auto_failover/src/bin'
gmake: *** [Makefile:127: bin] Error 2
```

--- a/src/bin/pg_autoctl/Makefile
+++ b/src/bin/pg_autoctl/Makefile
@@ -41,6 +41,9 @@ DEFAULT_CFLAGS += -Werror=implicit-function-declaration
 DEFAULT_CFLAGS += -Werror=return-type
 DEFAULT_CFLAGS += -Wno-declaration-after-statement
 
+# Needed for FreeBSD
+DEFAULT_CFLAGS += -D_WANT_SEMUN
+
 # Needed for OSX
 DEFAULT_CFLAGS += -Wno-missing-braces
 DEFAULT_CFLAGS += $(COMMON_LIBS)

